python/ray/_private/auto_init_hook.py

```diff
@@ -4,13 +4,11 @@
 import threading
 
 auto_init_lock = threading.Lock()
+enable_auto_connect = os.environ.get("RAY_ENABLE_AUTO_CONNECT", "") != "0"
 
 
 def auto_init_ray():
-    if (
-        os.environ.get("RAY_ENABLE_AUTO_CONNECT", "") != "0"
-        and not ray.is_initialized()
-    ):
+    if enable_auto_connect and not ray.is_initialized():
```

Comment:
+1

Beat me to fixing this. It's a small thing but notoriously showing up in every profile